examples/complete/main.tf

```diff
@@ -4,21 +4,20 @@ provider "aws" {
 
 module "vpc" {
   source     = "../../"
-  namespace  = var.namespace
-  stage      = var.stage
-  name       = var.name
   cidr_block = "172.16.0.0/16"
+
+  context = module.this.context
 }
 
 module "subnets" {
-  source               = "git::https://github.com/cloudposse/terraform-aws-dynamic-subnets.git?ref=tags/0.15.0"
+  source = "git::https://github.com/cloudposse/terraform-aws-dynamic-subnets.git?ref=tags/0.28.0"
+
   availability_zones   = var.availability_zones
-  namespace            = var.namespace
-  stage                = var.stage
-  name                 = var.name
   vpc_id               = module.vpc.vpc_id
   igw_id               = module.vpc.igw_id
   cidr_block           = module.vpc.vpc_cidr_block
   nat_gateway_enabled  = false
   nat_instance_enabled = false
+
+  context = module.this.context
```

Comment:
Remind me why you liked the context at the end? Dont' all the `vars.*` override the context anyways? 

Counter arguments:
- In languages where context is passed, it's usually the first argument (e.g. perl, javascript)
- Since terraform ascribes to putting `source` and `version` first (though I can't find the citation for this atm), I would put this third. 
- Another argument would be to put everything that always appears in an invocation of a module at the top.

Reply:
@osterman Having the source and version at the top makes sense because it is not boilerplate but rather key information that establishes what the module is, so you know what you are looking at. I want to put `context=` last because it is boilerplate and I like to have boilerplate last if i cannot hide it, so that I don't have to scroll through it to get to the interesting stuff.